internal/pkg/service/stream/sink/router/router.go

```diff
@@ -40,23 +47,43 @@ type Router struct {
 
 	// wg waits for all writes/goroutines
 	wg sync.WaitGroup
+
+	// OTEL metrics
+	meters *meters
+}
+
+type meters struct {
+	sourceDuration metric.Float64Histogram
+	sourceBytes    metric.Int64Histogram
+	sinkDuration   metric.Float64Histogram
+	sinkBytes      metric.Int64Histogram
 }
 
 type dependencies interface {
+	Clock() clock.Clock
 	Logger() log.Logger
 	Process() *servicectx.Process
 	Plugins() *plugin.Plugins
 	DefinitionRepository() *definitionRepo.Repository
+	Telemetry() telemetry.Telemetry
 }
 
-func New(d dependencies) (*Router, error) {
+func New(d dependencies, sourceType string) (*Router, error) {
 	r := &Router{
+		sourceType:  sourceType,
+		clock:       d.Clock(),
 		logger:      d.Logger().WithComponent("sink.router"),
 		plugins:     d.Plugins(),
 		definitions: d.DefinitionRepository(),
 		collection:  newCollection(),
 		closed:      make(chan struct{}),
 		pipelines:   make(map[key.SinkKey]*pipelineRef),
+		meters: &meters{
+			sourceDuration: d.Telemetry().Meter().FloatHistogram("keboola.go.stream.source.in.duration", "Duration of source requests.", "ms"),
+			sourceBytes:    d.Telemetry().Meter().IntHistogram("keboola.go.stream.source.in.bytes", "Source request length.", "B"),
+			sinkDuration:   d.Telemetry().Meter().FloatHistogram("keboola.go.stream.sink.in.duration", "Duration of source requests dispatched to sink.", "ms"),
+			sinkBytes:      d.Telemetry().Meter().IntHistogram("keboola.go.stream.sink.in.bytes", "Bytes written to sink.", "B"),
```

Comment:
:thinking: 

`keboola.go.stream.source.in.duration` -> `keboola.go.stream.source.{x/rec/record}.duration`

`keboola.go.stream.sink.in.duration` -> `keboola.go.stream.sink.{x/rec/record}.duration`
`keboola.go.stream.sink.in.bytes` -> `keboola.go.stream.sink.{out/encoded/uncompressed/raw}.bytes`